src/main/java/org/cactoos/scalar/ScalarWithFallback.java

```diff
@@ -103,16 +105,18 @@ public T value() throws Exception {
      */
     @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
     private T fallback(final Throwable exp) throws Exception {
-        final Sorted<Map.Entry<Fallback<T>, Integer>> candidates =
+        final Iterator<Map.Entry<Fallback<T>, Integer>> candidates =
             new Sorted<>(
                 Comparator.comparing(Map.Entry::getValue),
                 new Filtered<>(
-                    entry -> new Not(
-                        new Equals<>(
-                            entry::getValue,
-                            () -> Integer.MIN_VALUE
+                    new org.cactoos.func.Flattened<>(
+                        entry -> new Not(
+                            new Equals<Integer, Integer>(
+                                entry::getValue,
+                                new Constant<>(Integer.MIN_VALUE)
+                            )
                         )
-                    ).value(),
+                    ),
```

Comment:
@victornoel Just to know. What is the real value that `Flattened` brings here compared to the last code ?

Reply:
@baudoliver7 one of the driving idea behind EO is that you compose objects without calling methods (i.e., pure declarativeness). So a lot of changes happening to cactoos go into that direction: using `Flattened` delegates the responsibility of actually calling the `value()` method to an object who has this sole responsibility. It may seem a bit exaggerated, but well, cactoos has always been an experimentation field for extremist OO and EO :)

I'm still not 100% sure if it's always best, but let's see how it ends up :) `Ternary` is also a good example of such extremism, both in its definition but also in its usage composed with `Flattened` for example.

Reply:
@victornoel Thank you for your explanation :)